prov/shm/src/smr_ep.c

```diff
@@ -1062,10 +1062,16 @@ static struct fi_ops smr_ep_fi_ops = {
 	.ops_open = fi_no_ops_open,
 };
 
-static void smr_endpoint_name(char *name, int pid, int dom_idx, int ep_idx)
-{	
+static void smr_endpoint_name(char *name, char *addr, size_t addrlen,
+			      int pid, int dom_idx, int ep_idx)
+{
 	memset(name, 0, SMR_NAME_SIZE);
-	snprintf(name, SMR_NAME_SIZE, "%d:%d:%d", pid, dom_idx, ep_idx);
+	if (addr) {
+		snprintf(name, addrlen < SMR_NAME_SIZE ?
+			 addrlen : SMR_NAME_SIZE, "%s", addr);
```

Comment:
You probably want to fail if the addrlen is longer than the supported length, rather than silently truncating the address